cmd/crank/beta/validate/cmd.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+)
+
+// Cmd arguments and flags for render subcommand.
+type Cmd struct {
+	// Arguments.
+	Extensions string `arg:"" help:"Extensions source which can be a file, directory, or '-' for standard input."`
+	Resources  string `arg:"" help:"Resources source which can be a file, directory, or '-' for standard input."`
+
+	// Flags. Keep them in alphabetical order.
+	SkipSuccessResults bool   `help:"Skip printing success results."`
+	CacheDir           string `help:"Absolut path to the cache directory where downloaded schemas are stored." default:".crossplane/cache"`
+	CleanCache         bool   `help:"Clean the cache directory before downloading package schemas."`
+
+	fs afero.Fs
+}
+
+// Help prints out the help for the render command.
+func (c *Cmd) Help() string {
+	return `
+This command validates the provided Crossplane resources against the schemas of the provided extensions (e.g., XRDs and 
+CRDs, as well as Providers and Configurations coming soon). The output of the "crossplane beta render" command can be 
+piped to this validate command in order to rapidly validate on the outputs of the composition development experience.
+
+All validation is performed offline locally using the Kubernetes API server's validation library, so it does not require 
+any Crossplane instance or control plane to be running or configured.
+
+Examples:
+
+  # Validate all resources in the resources.yaml file against the extensions in the extensions.yaml file
+  crossplane beta validate extensions.yaml resources.yaml
+
+  # Validate all resources in the resourceDir folder against the extensions in the extensionsDir folder and skip 
+  # success logs
+  crossplane beta validate extensionsDir/ resourceDir/ --skip-success-logs
+ 
+  # Validate the output of the render command against the extensions in the extensionsDir folder
+  crossplane beta render xr.yaml composition.yaml func.yaml | crossplane beta validate extensionsDir/ -
+`
+}
+
+// AfterApply implements kong.AfterApply.
+func (c *Cmd) AfterApply() error {
+	c.fs = afero.NewOsFs()
+	return nil
+}
+
+// Run validate.
+func (c *Cmd) Run(_ *kong.Context, _ logging.Logger) error {
+	// Load all extensions
+	extensionLoader, err := NewLoader(c.Extensions)
+	if err != nil {
+		return errors.Wrapf(err, "cannot load extensions from %q", c.Extensions)
+	}
+
+	extensions, err := extensionLoader.Load()
+	if err != nil {
+		return errors.Wrapf(err, "cannot load extensions from %q", c.Extensions)
+	}
+
+	// Load all resources
+	resourceLoader, err := NewLoader(c.Resources)
+	if err != nil {
+		return errors.Wrapf(err, "cannot load resources from %q", c.Resources)
+	}
+
+	resources, err := resourceLoader.Load()
+	if err != nil {
+		return errors.Wrapf(err, "cannot load resources from %q", c.Resources)
+	}
+
+	// Update default cache directory to absolute path based on the current working directory
+	if c.CacheDir == defaultCacheDir {
+		currentPath, err := os.Getwd()
```

Comment:
We already create a .crossplane folder in the home, we could just reuse that: https://github.com/crossplane/crossplane/blob/a5c2971a2e06260751765ad91c54da1016df2dd3/internal/xpkg/upbound/config/config.go#L36

Another thing we might want to address in a followup PR, if so, might be worth adding a TODO.